skbio/io/fasta.py

```diff
@@ -557,6 +563,10 @@
 
 @register_sniffer('fasta')
 def _fasta_sniffer(fh):
+
+    if _count_blank_lines(fh) > 5:
```

Comment:
@jairideout I don't think there's a simple way to move the `max_blank` logic up to `_line_generator`. The problem is that the sniffer doesn't use `_line_generator` directly, so there's no where to capture a return value and indicate the file was not identified by the sniffer. So the only way I thought to do it was have `_line_generator` raise an exception and have the sniffer catch it. But then the files always except, not just when being sniffed. I haven't been able to come up with a way to branch that logic.